worldedit-bukkit/src/main/java/com/fastasyncworldedit/bukkit/adapter/IBukkitAdapter.java

```diff
@@ -186,10 +188,12 @@ default Material adapt(BlockType blockType) {
     }
 
     default org.bukkit.entity.EntityType adapt(EntityType entityType) {
-        if (!entityType.getId().startsWith("minecraft:")) {
-            throw new IllegalArgumentException("Bukkit only supports vanilla entities");
+        NamespacedKey entityKey = NamespacedKey.fromString(entityType.toString());
+        if (entityKey == null) {
+            throw new IllegalArgumentException("Entity key '" + entityType + "' does not map to Bukkit");
         }
-        return org.bukkit.entity.EntityType.fromName(entityType.getId().substring(10).toLowerCase(Locale.ROOT));
+
+        return Registry.ENTITY_TYPE.get(entityKey);
```

Comment:
Original https://github.com/EngineHub/WorldEdit/blob/79a06b1d21bacfb8b6ab1df1c2452da119fdddf4/worldedit-bukkit/src/main/java/com/sk89q/worldedit/bukkit/BukkitAdapter.java#L399-L406
(EntityType is not an record on our end - which theoretically breaks compatibility - so I've used toString instead of the id accessor. Might be worth adding that method to simulate the record behavior?)